pyccel/codegen/printing/ccudacode.py

```diff
@@ -521,19 +521,25 @@ def copy_CudaArray_Data(self, expr):
         declare_dtype = self.find_in_dtype_registry(self._print(rhs.dtype), rhs.precision)
         dtype = self.find_in_ndarray_type_registry(self._print(rhs.dtype), rhs.precision)
         arg = rhs.arg if isinstance(rhs, (CudaArray, CupyArray)) else rhs
+
+        memcpy_kind_src = str(rhs.current_context).capitalize()
+        memcpy_kind_dest = 'Host' if rhs.memory_location == 'host' else 'Device'
```

Comment:
Could this be:
```python
        memcpy_kind_src = str(rhs.current_context).capitalize()
        memcpy_kind_dest = rhs.memory_location.capitalize()
```
?
I'm not sure why you're using 2 different methods here.

Reply:
Oh that's to not have a "Managed" `cudaMemcpy` destination, since the only valid value are "Host" and "Device", and it name imply the `managed` memory location can work with any of those two values.

Reply:
Ok perfect :+1: